### DISCUSSION TITLE
Share sensor across devices

### DISCUSSION THREAD

**Discussion by Freebios** [dawidchyrzynski/arduino-home-assistant#237] [Q&A] [answered]
Hello there !

I would like to know if there is straightforward way of sharing a sensor from a device A with a device B?

I run a bunch of esp32 gathering data and publishing it to HA. I would like physical device A to be able to react to changes in device B sensors values by subscribing to its topics.

Is this something achievable by instantiating a sensor class with the same id on the two devices? I don't seem to be able to get it work (it would make sense since it would be hashed with device unique id)

Otherwise I assume I would need to listen to the topic as described in the advanced mqtt documentation.

Thanks for your feedback!

**Comment by dawidchyrzynski** [comment, marked as answer]
Hi @Freebios,

At present, the sole method to achieve this task is by manually subscribing to the topics as shown in the advanced documentation. I will consider implementing an additional class that would streamline this process.

**Reply by Freebios** [reply]
Thanks for the feedback, I'll go along that path for now, and thanks a lot for maintaining this project :)
